pyaerocom/aeroval/modelmaps_engine.py

```diff
@@ -137,8 +143,20 @@ def _process_map_var(self, model_name, var, reanalyse_existing):
                 logger.info(f"Skipping processing of {outname}: data already exists.")
                 return []
 
-        freq = self.cfg.time_cfg.main_freq
+        mainfreq = self.cfg.time_cfg.main_freq
+        allfreqs = self.cfg.time_cfg.freqs
         tst = TsType(data.ts_type)
+        allfreqs = [TsType(fq) for fq in freqs]  # convert them into TsType objects
+        allfreqs.sort(
+            reverse=True
+        )  # sort them from least coarse to coarsest, e.g. [hourly, daily, monthly]
+        coarsestfreq = allfreqs[-1]
+
+        if mainfreq < coarsestfreq:
+            freq = mainfreq
+        else:
+            freq = coarsestfreq
```

Comment:
It looks to me that  `allfreqs` is only defined to calculate `coarsestfreq`.
If this is the case you could write as follows:
``` python
        coarsestfreq = min(TsType(fq) for fq in self.cfg.time_cfg.freqs)
```

Also, is it possible for `self.cfg.time_cfg.main_freq` not to be included in `self.cfg.time_cfg.freqs`? In this case, this calculation could be rewritten as:
``` python
        freq = min(TsType(fq) for fq in self.cfg.time_cfg.freqs)
        freq = min(freq, self.cfg.time_cfg.main_freq)
```

Reply:
good point, I assumed not, but it's safer maybe to assume it could